io.github.shadowmario.fnf-psychengine.yml

```diff
@@ -0,0 +1,87 @@
+        config-opts:
+          - -Dwith-sd-bus-provider=no-daemon
+          - -Dwith-examples=false
+        sources:
+          - type: git
+            url: https://github.com/FeralInteractive/gamemode
+            tag: "1.8.1"
+            x-checker-data:
+            type: git
+            tag-pattern: ^([\\d.]+)$
+    sources:
+      - type: archive
+        url: https://github.com/HaxeFoundation/neko/releases/download/v2-3-0/neko-2.3.0-linux64.tar.gz
+        sha256: 26dda28d0a51407f26218ba9c2c355c8eb23cf2b0b617274b00e4b9170fe69eb
+        dest: haxe/
+        x-checker-data:
+          type: anitya
+          project-id: 10885
+          url-template: https://github.com/HaxeFoundation/neko/releases/download/v$version/neko-$version-linux64.tar.gz
+      - type: archive
+        url: https://github.com/HaxeFoundation/haxe/releases/download/4.3.3/haxe-4.3.3-linux64.tar.gz
+        sha256: ee7dbdad1dbceb0f36a2453cfe36a1310a1b7df4f134f6a8d5d77951edaa18c8
+        dest: haxe/
+        x-checker-data:
+          type: anitya
+          project-id: 7883
+          url-template: https://github.com/HaxeFoundation/haxe/releases/download/$version/haxe-$version-linux64.tar.gz
+      - type: git
+        url: https://github.com/ShadowMario/FNF-PsychEngine.git
+        # tag: "0.7.3"
+        branch: main
+        commit: d64912e57120e27dc41304deae48097b6c1355d1
+        x-checker-data:
+          type: git
+          tag-pattern: ^([\\d.]+)$
+      # temporary fix for broken build in the repo (https://github.com/ShadowMario/FNF-PsychEngine/pull/14170)
+      - type: patch
+        path: fnf-psychengine-fixbuild.patch
+      # redirect crash logs, mods directory and list to XDG_DATA_HOME
+      - type: patch
+        path: fnf-psychengine-fix_user_data.patch
+      # launcher script
+      - type: script
+        dest-filename: io.github.shadowmario.fnf-psychengine.sh
+        commands:
+          # Launch game
+          - cd /app/bin
+          - ./PsychEngine
+    builddir: true
+    buildsystem: simple
+    build-commands:
+      # build
+      - |
+        export PATH=$PATH:$(pwd)/haxe
+        haxe -cp ./setup -D analyzer-optimize -main Main --interp
+        ./haxe/haxelib run lime build Project.xml linux
+        mkdir -p /app/bin && cp -r export/release/linux/bin/* /app/bin/
+      # Install launchers
+      - |
+        install -m755 io.github.shadowmario.fnf-psychengine.sh -t /app/bin/
+        desktop-file-edit --set-key=Exec --set-value=io.github.shadowmario.fnf-psychengine.sh assets/meta/io.github.shadowmario.fnf-psychengine.desktop
+        install -Dm 644 assets/meta/io.github.shadowmario.fnf-psychengine.desktop -t /app/share/applications
+        install -Dm 644 assets/meta/io.github.shadowmario.fnf-psychengine.png -t /app/share/icons/hicolor/512x512/apps
+        install -Dm 644 assets/meta/io.github.shadowmario.fnf-psychengine.metainfo.xml -t /app/share/metainfo
+    build-options:
+      build-args:
+        - --share=network
```

Comment:
this is not allowed on the builders.  Build must be offline

Reply:
I had included this because the build pulls libraries for the development toolkit from an online repository at build time (much like npm). I've untangled those dependencies and added them as separate explicit sources but there's over a dozen of them - is it possible to have a list of sources in a separate yaml file to keep the main build manifest a bit tidier, or can only entire modules be offloaded to external files like shared modules?